HW_module_8.py

```diff
@@ -0,0 +1,151 @@
+                        with open(file_name, "r", encoding='utf-8') as read_file:
+                            lict_of_dicts = json.load(read_file)
+                        print('Such file exists')
+                        path_for_remove = os.path.join(str(change_path), str(file_name))
+                        os.chdir(original_path)
+                        is_true = False
+                        break
+                    except FileNotFoundError:
+                        print('No file with such name. Try again')
+                    except PermissionError:
+                        print('No file with such name. Try again')
+            else:
+                print('Try again')
+        return lict_of_dicts, path_for_remove
+
+    def publishing(self):
+        lict_of_dicts, path_for_remove = self.read_file()
+
+        insert_into_file = []
+        try:
+            for i, d in enumerate(lict_of_dicts):
+                data = []
+                if d["type"] == 'News':
+                    if len(d["text"]) > 0 and len(d["location"]) > 0 and len(d["date"]) > 0:
+                        data.append('News ----------------------------'+'\n')
+                        data.append(d["text"] + "\n")
+                        data.append(d["location"] + ', ' + str(d["date"]) + "\n")
+                        data.append('---------------------------------' + "\n\n\n")
+                        insert_into_file.append(data)
+                elif d["type"] == 'Private Ad':
+                    if len(d["text"]) > 0 and len(d["date"]) > 0:
+                        data.append('Private Ad -----------------------'+'\n')
+                        data.append(d["text"] + "\n")
+                        data.append(str(d["date"]) + "\n")
+                        data.append('---------------------------------' + "\n\n\n")
+                        insert_into_file.append(data)
+                elif d["type"] == 'How do you think?':
+                    if len(d["question"]) > 0 and len(d["answer"]) > 0 and len(d["conclusion"]) > 0:
+                        data.append('How do you think? ----------------'+'\n')
+                        data.append(d["question"] + "\n")
+                        data.append(d["answer"] + "\n")
+                        data.append(d["conclusion"] + "\n")
+                        data.append('---------------------------------' + "\n\n\n")
+                        insert_into_file.append(data)
+                else:
+                    print('Some unknown type of records was found')
+                    insert_into_file = 'Empty'
+                    break
+        except KeyError:
+            print('Some important field is absent for block #', i + 1, sep='')
+            insert_into_file = 'Empty'
+
+        return insert_into_file, path_for_remove
+
+
+# ask a user what data he wants to print and then call a class and insert the data into file using inserting method
+if __name__ == "__main__":
```

Comment:
This logic must be presented in a single file, class or anything else. No need to copy the logic of your application for all files.